### PR TITLE
storage: Resolve EBS flashback unlimit range

### LINKED ISSUE #13879: Flashback may miss the last region
### What version of TiKV are you using?

https://github.com/tikv/tikv/commit/4b4fc4390d90e72eb87e49a223e756e6ba1e2688

### What happened?

`end_key` in a flashback request will be encoded directly without checking whether it's empty.

https://github.com/tikv/tikv/blob/4b4fc4390d90e72eb87e49a223e756e6ba1e2688/src/storage/txn/commands/mod.rs#L369-L379

Inside the flashback progress, if the `end_key` is empty, it will be less than all other keys, which will cause `scan_latest_user_keys` to return immediately without flashbacking any keys.

https://github.com/tikv/tikv/blob/4b4fc4390d90e72eb87e49a223e756e6ba1e2688/src/storage/mvcc/reader/reader.rs#L611-L616

Luckily, this won't be a problem for the Flashback SQL since it will use the table prefix key as the `end_key`.

https://github.com/pingcap/tidb/blob/0b1096eac5a500f8c624f08f384d0194da5386f4/ddl/cluster.go#L255-L258

### PR DESCRIPTION
**Pull request by HuSharp**
Signed-off-by: husharp <jinhao.hu@pingcap.com>

### What is changed and how it works?

Issue Number: Close #13879 

What's Changed:

```commit-message
Resolve EBS flashback unlimit range.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -4875,7 +4875,7 @@ mod tests {
                 commit_ts,
                 version,
                 key.clone(),
-                Key::from_raw(b"z"),
+                Key::from_raw_maybe_unbounded(b"z"),
```
Comment (JmPotato) [inline src/storage/mod.rs:4878]: What about passing `Some(Key::from_raw(b"z"))` in tests since we can make sure that it's not empty?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: src/storage/mod.rs (hunk 1 above)
Comment (HuSharp) [reply src/storage/mod.rs:4878]: Okay, get it.

**Review by sticnarf — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Comment by HuSharp** [comment]
@sticnarf can u merge this pr? Thx!!!

**Pull request merged**